src/rpc/daemon_handler.cpp

```diff
@@ -0,0 +1,881 @@
+    }
+
+    NOTIFY_NEW_TRANSACTIONS::request r;
+    r.txs.push_back(tx_blob);
+    m_core.get_protocol()->relay_transactions(r, fake_context);
+
+    //TODO: make sure that tx has reached other nodes here, probably wait to receive reflections from other nodes
+    res.status = Message::STATUS_OK;
+    res.relayed = true;
+
+    return;
+  }
+
+  void DaemonHandler::handle(const StartMining::Request& req, StartMining::Response& res)
+  {
+    account_public_address adr;
+    if(!get_account_address_from_str(adr, m_core.get_testnet(), req.miner_address))
+    {
+      res.error_details = "Failed, wrong address";
+      LOG_PRINT_L0(res.error_details);
+      res.status = Message::STATUS_FAILED;
+      return;
+    }
+
+    unsigned int concurrency_count = boost::thread::hardware_concurrency() * 4;
+
+    // if we couldn't detect threads, set it to a ridiculously high number
+    if(concurrency_count == 0)
+    {
+      concurrency_count = 257;
+    }
+
+    // if there are more threads requested than the hardware supports
+    // then we fail and log that.
+    if(req.threads_count > concurrency_count)
+    {
+      res.error_details = "Failed, too many threads relative to CPU cores.";
+      LOG_PRINT_L0(res.error_details);
+      res.status = Message::STATUS_FAILED;
+      return;
+    }
+
+    boost::thread::attributes attrs;
+    attrs.set_stack_size(THREAD_STACK_SIZE);
+
+    if(!m_core.get_miner().start(adr, static_cast<size_t>(req.threads_count), attrs, req.do_background_mining, req.ignore_battery))
+    {
+      res.error_details = "Failed, mining not started";
+      LOG_PRINT_L0(res.error_details);
+      res.status = Message::STATUS_FAILED;
+      return;
+    }
+    res.status = Message::STATUS_OK;
+    res.error_details = "";
+
+  }
+
+  void DaemonHandler::handle(const GetInfo::Request& req, GetInfo::Response& res)
+  {
+    res.info.height = m_core.get_current_blockchain_height();
+
+    res.info.target_height = m_core.get_target_blockchain_height();
+
+    if (res.info.height > res.info.target_height)
+    {
+      res.info.target_height = res.info.height;
+    }
+
+    auto& chain = m_core.get_blockchain_storage();
+
+    res.info.difficulty = chain.get_difficulty_for_next_block();
+
+    res.info.target = chain.get_difficulty_target();
+
+    res.info.tx_count = chain.get_total_transactions() - res.info.height; //without coinbase
+
+    res.info.tx_pool_size = m_core.get_pool_transactions_count();
+
+    res.info.alt_blocks_count = chain.get_alternative_blocks_count();
+
+    uint64_t total_conn = m_p2p.get_connections_count();
+    res.info.outgoing_connections_count = m_p2p.get_outgoing_connections_count();
+    res.info.incoming_connections_count = total_conn - res.info.outgoing_connections_count;
+
+    res.info.white_peerlist_size = m_p2p.get_peerlist_manager().get_white_peers_count();
+
+    res.info.grey_peerlist_size = m_p2p.get_peerlist_manager().get_gray_peers_count();
+
+    res.info.testnet = m_core.get_testnet();
+
```

Comment:
I'll go through all the calls to make sure they're up to date, and get rid of the deprecated ones.